recipes/aeron/all/conanfile.py

```diff
@@ -0,0 +1,137 @@
+import os
+import shutil
+import glob
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+
+class AeronConan(ConanFile):
+    name = "aeron"
+    description = "Efficient reliable UDP unicast, UDP multicast, and IPC message transport"
+    topics = ("conan", "aeron", "udp", "messaging", "low-latency")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/real-logic/aeron/wiki"
+    license = "Apache-2.0"
+    exports_sources = "CMakeLists.txt",
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "build_aeron_driver": [True, False],
+        "build_aeron_archive_api": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "build_aeron_driver": True,
+        "build_aeron_archive_api": True
+    }
+    generators = "cmake"
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+        compiler = str(self.settings.compiler)
+        compiler_version = tools.Version(self.settings.compiler.version)
+
+        minimal_version = {
+            "Visual Studio": "16",
+            "gcc": "5"
+        }
+
+        if compiler in minimal_version and compiler_version < minimal_version[compiler]:
+            raise ConanInvalidConfiguration(
+                "{} requires {} compiler {} or newer [is: {}]".format(self.name, compiler, minimal_version[compiler], compiler_version)
+            )
```

Comment:
```suggestion
        compiler = str(self.settings.compiler)
        compiler_version = tools.Version(self.settings.compiler.version)

        minimal_version = {
            "Visual Studio": "16",
            "gcc": "5"
        }

        if compiler in minimal_version and compiler_version < minimal_version[compiler]:
            raise ConanInvalidConfiguration(
                "{} requires {} compiler {} or newer [is: {}]".format(self.name, compiler, minimal_version[compiler], compiler_version)
            )
```

Move it to validate() method. C++11 doesn't require MSVC 16. Take a look https://en.cppreference.com/w/cpp/compiler_support

Reply:
@uilianries  `validate` is an experimental method -  https://docs.conan.io/en/latest/reference/conanfile/methods.html#validate A warning in the box clearly says it might change in the future, thus I don't think it's a good idea to use it. 

Additionally, other packages also use `configure` to indicate unsupported compiler versions. A few examples:
* **folly** - https://github.com/conan-io/conan-center-index/blob/master/recipes/folly/all/conanfile.py#L50
* **fruit** - https://github.com/conan-io/conan-center-index/blob/master/recipes/fruit/all/conanfile.py#L48
* **libpqxx** - https://github.com/conan-io/conan-center-index/blob/master/recipes/libpqxx/all/conanfile.py#L41

Aeron requires at least Visual Studio 2017 (15) to build media driver - https://github.com/real-logic/aeron/blob/master/aeron-driver/src/main/c/README.md. Also some of the definitions in [https://github.com/real-logic/aeron/blob/master/aeron-client/src/main/cpp/concurrent/atomic/Atomic64_msvc.h](https://github.com/real-logic/aeron/blob/master/aeron-client/src/main/cpp/concurrent/atomic/Atomic64_msvc.h) are based on [_InterlockedExchangeAdd64](https://docs.microsoft.com/en-us/windows/win32/api/winnt/nf-winnt-interlockedexchangeadd64) which was introduced in **winnt.h** released with VS 2017. For VS2017 we observed some other issues which you can lookup here: [https://c3i.jfrog.io/c3i/misc/logs/pr/5035/14/aeron/1.31.2//1ab8c05892c9fc629b34f483da1c50a905e6d956-build.txt](https://c3i.jfrog.io/c3i/misc/logs/pr/5035/14/aeron/1.31.2//1ab8c05892c9fc629b34f483da1c50a905e6d956-build.txt).

I'm more than happy to change minimal version of Visual Studio to 15, if you propose a solution which mitigates those build issues.

Reply:
> `validate` is an experimental method

uilianries is a part of the Conan team 😉 Also from what I gather, Conan leverages CCI to prove it's bleeding cutting edge super awesome new features... Validate has been available for many months, I have not seen any issue that would warnet it to change. Those example you picked have not been touched since september (except folly which is a nightmare)... there are lots of places we are using validate.

➡️ It's very acceptable to use the validate.

> I'm more than happy to change minimal version of Visual Studio to 15, if you propose a solution which mitigates those build issues.

The only error I fond was

```
MSBUILD : error MSB4166: Child node "9" exited prematurely. Shutting down. Diagnostic information may be found in files in "C:\J\w\BuildSingleReference@2\9148\11cd7489-7eb7-41ca-b2b3-22d3a12e0ed2\" and will be named MSBuild_*.failure.txt. This location can be changed by setting the MSBUILDDEBUGPATH environment variable to a different directory.
```

Which is from the build system running out of memory when building AFAIK.... it's rare but not new. It can happen to any windows package build.

When you see that error you can retrigger CI by closing the pr waiting 10s before re-opening it 🔁

Reply:
No problem. I moved the checks to `validate`.